packages/components/src/internal/util/utils.ts

```diff
@@ -329,11 +339,11 @@ export function getUpdatedData(
                     });
 
                     return m.set(key, updatedVal);
-                } else if (updateValuesMap.has(encodedKey) && updatedVal === undefined) {
+                } else if (updateValuesMap.has(col.fieldKey) && updatedVal === undefined) {
                     return m.set(key, []);
                 } else return m;
             } else return m;
-        }, Map<any, any>());
+        }, Map<string, any>());
```

Comment:
The function `getUpdatedRowForParentChanges()` seems suspect for it's usage of `encodePart` in its preparation of rows for update. Is this something that should fall under this PR? Or potentially a follow-on?

Reply:
I believe those particular usages of `encodePart` are expected since `getUpdatedRowForParentChanges` is dealing with sample type / source type names and generating a multi part fieldKey. That said, I see there are a couple other usages of `encodePart` in ui-premium that I need to review to see if they can be removed. I'll do that as a separate PR at this point though.